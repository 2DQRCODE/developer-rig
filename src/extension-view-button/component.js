import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className="circle-button" onClick={this.props.onClick}>
        <p>+</p>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
