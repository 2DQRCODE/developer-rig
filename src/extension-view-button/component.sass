.circle-button
  border-width: thick
  border-radius: 50%
  width: 50px
  height: 50px
  margin: 0 auto
  margin-top: 10px
  border: 2px solid #DAD8DE
  position: relative
  font-weight: bold
  display: table
  cursor: pointer

.circle-button p
  text-align: center
  vertical-align: middle
  display: table-cell

