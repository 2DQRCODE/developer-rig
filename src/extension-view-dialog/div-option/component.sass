@import '../component.sass'
$dimension: 70px

.extension-container__block
  @extend .container__block

  .block__extension-type-label
    visibility: hidden
    position: absolute
    cursor: pointer

  .extension-type-label__extension-type-text--color-black
    @extend .extension-type-text

  .extension-type-label__extension-type-text--color-purple
    @extend .extension-type-text
    color: #7D55C7

.extension-type-box
  margin-top: 10px
  margin-right: 12px
  height: $dimension
  width: $dimension
  border-radius: 5px
  border: 2px solid black
  display: table-cell
  vertical-align: middle
  text-align: center

.extension-type-text
    margin: 3px
    font-size: 10px
    text-align: center
    width: $dimension
