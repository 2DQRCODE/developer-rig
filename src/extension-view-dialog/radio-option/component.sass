.option-label
  clear: left
  font-size: 12px
  margin-right: 10px
