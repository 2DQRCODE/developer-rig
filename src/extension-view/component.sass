.view
  border: 1px solid #D3D0D6
  margin-left: auto
  margin-right: auto

.view__wrapper
  padding: 2em
  padding-top: 0
  border-left: 1px solid #DAD8DE
  border-right: 1px solid #DAD8DE
  border-bottom: 1px solid #DAD8DE
  flex-grow: 1
  flex-basis: auto
  display: flex
  flex-direction: column

.view__header
  padding: 1em
  height: 20px

.view__close_button
  float: right
  padding-left: .8em
  color: #511798
  cursor: pointer

.view__descriptor
  float: left
  padding-right: .8em
  font-size: .75em
  color: #A4A4A4

.rig-frame
  flex-grow: 1
