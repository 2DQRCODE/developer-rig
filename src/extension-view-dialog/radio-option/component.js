import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class RadioOption extends Component {
  render() {
    return (
      <div>
        <input type="radio" name={this.props.name} value={this.props.value}  onChange={this.props.onChange} defaultChecked={this.props.checked}/>
        <label className="option-label"> {this.props.value} </label>
      </div>
    )
  }
}

RadioOption.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool
}
