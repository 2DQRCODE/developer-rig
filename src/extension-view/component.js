import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
import { ExtensionFrame } from '../extension-frame';
import { IdentityOptions } from '../constants/identity-options';
import { ViewerTypes } from '../constants/viewer_types';
import closeButton from '../img/close_icon.png';
const { ExtensionAnchor, ExtensionMode } = window['extension-coordinator'];

export class ExtensionView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      mousedOver: false,
    };
  }

  mouseEnter() {
    this.setState({
      mousedOver: true,
    });
  }

  mouseLeave() {
    this.setState({
      mousedOver: false
    });
  }

  renderLinkedOrUnlinked() {
    return this.props.linked ? IdentityOptions.Linked : IdentityOptions.Unlinked;
  }

  render() {
    const extensionProps = {}
    switch(this.props.type) {
      case ExtensionAnchor.Panel:
        extensionProps.viewStyles = {
          width: "320px",
        };
        break;
      case ExtensionAnchor.Overlay:
        extensionProps.viewStyles = {
          width: this.props.overlaySize.width,
          height: this.props.overlaySize.height
        };
        break;
      case ExtensionMode.Config:
        extensionProps.viewStyles = {
          width: "100%",
          height: "700px"
        };
        extensionProps.viewWrapperStyles = {
          overflow: "auto",
          height: "70vh",
        };
        break;
      default:
        break;
    }

    return (

      <div
        className={'view__wrapper'}
        onMouseEnter={() => { this.mouseEnter() }}
        onMouseLeave={() => { this.mouseLeave() }}
        style={extensionProps.viewWrapperStyles}>
        <div
          className={'view__header'}>
          {(this.props.deleteViewHandler !== undefined && this.state.mousedOver) &&
            (<div className={'view__close_button'}
              onClick={() => { this.props.deleteViewHandler(this.props.id) }}>
            <img
              alt='Close'
              src={closeButton}
            />
            </div>)
          }
          <div className={'view__descriptor'}>
            { this.props.role }
          </div>
          <div className={'view__descriptor'}>
            {(this.props.role === ViewerTypes.LoggedIn) ?
              this.renderLinkedOrUnlinked() : null}
          </div>
        </div>
        <div
          className="view"
          ref={this._boundIframeHostRefHandler}
          style={extensionProps.viewStyles}>
          <ExtensionFrame
            iframeClassName={`rig-frame frameid-${this.props.id}`}
            extension={this.props.extension}
            type={this.props.type}
            mode={this.props.mode}
          />
        </div>
      </div>
    );
  }
}

ExtensionView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  role: PropTypes.string,
  overlaySize: PropTypes.object,
};
