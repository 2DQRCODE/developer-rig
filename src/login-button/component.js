import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class LoginButton extends Component {
  render() {
    return (
      <div className="login-button">
        <button
          onClick={this.props.loginHandler}>Log In</button>
      </div>
    );
  }
}

LoginButton.propTypes = {
  loginHandler: PropTypes.func.isRequired
};
