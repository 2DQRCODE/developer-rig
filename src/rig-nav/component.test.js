import React from 'react';
import { shallow } from 'enzyme';
import { RigNav } from './component';
import { EXTENSION_VIEWS } from '../constants/nav-items';

describe('<RigNav />', () => {
  describe('receives an error', () => {
    const error = "RigNav Error";
    const component = shallow(
      <RigNav 
        error={error}/>
    );
  
    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });

    it('has the correct prop', () => {
      expect(component.instance().props.error).toEqual(error);
    });
  });

  describe('starts up successfully', () => {
    let usage = 0;
    const error = '';
    const func = () => {
      usage++;
    }
    const component = shallow(
      <RigNav
        openConfigurationsHandler={func}
        viewHandler={func}
        configHandler={func}
        liveConfigHandler={func}
        selectedView={EXTENSION_VIEWS}
      />
    );

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });

    it('has the correct props', () => {
      expect(component.instance().props.selectedView).toEqual(EXTENSION_VIEWS);
      expect(component.instance().props.viewHandler).toEqual(func);
      expect(component.instance().props.openConfigurationsHandler).toEqual(func);
      expect(component.instance().props.configHandler).toEqual(func);
      expect(component.instance().props.liveConfigHandler).toEqual(func);
    });
    
    it('handles clicks properly', () => {
      //TODO: implement
    });
  })
});
