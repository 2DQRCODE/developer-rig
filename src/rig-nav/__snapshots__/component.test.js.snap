// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigNav /> receives an error renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNav
    error="RigNav Error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a>
         
        RigNav Error
         
</a>,
      "className": "top-nav-error",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "RigNav Error",
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        "RigNav Error",
        " ",
      ],
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a>
           
          RigNav Error
           
</a>,
        "className": "top-nav-error",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "RigNav Error",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "RigNav Error",
          " ",
        ],
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RigNav /> starts up successfully renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNav
    configHandler={[Function]}
    liveConfigHandler={[Function]}
    openConfigurationsHandler={[Function]}
    selectedView="Extension Views"
    viewHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a>
         
         
</a>,
      "className": "top-nav-error",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          undefined,
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        undefined,
        " ",
      ],
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a>
           
           
</a>,
        "className": "top-nav-error",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
